contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/writers/HDF5LongDataWriter.java

```diff
@@ -0,0 +1,103 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.hdf5.writers;
+
+import ch.systemsx.cisd.hdf5.IHDF5Reader;
+import org.apache.drill.common.types.TypeProtos;
+import org.apache.drill.exec.physical.resultSet.RowSetLoader;
+import org.apache.drill.exec.store.hdf5.HDF5Utils;
+import org.apache.drill.exec.vector.accessor.ScalarWriter;
+
+import java.util.List;
+
+public class HDF5LongDataWriter extends HDF5DataWriter {
+
+  private final long[] data;
+
+  private final ScalarWriter rowWriter;
+
+  // This constructor is used when the data is a 1D column.  The column is inferred from the datapath
+  public HDF5LongDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String datapath) {
+    super(reader, columnWriter, datapath);
+    data = reader.readLongArray(datapath);
+
+    fieldName = HDF5Utils.getNameFromPath(datapath);
+    rowWriter = makeWriter(columnWriter, fieldName, TypeProtos.MinorType.BIGINT, TypeProtos.DataMode.OPTIONAL);
+
+  }
+
+  // This constructor is used when the data is part of a 2D array.  In this case the column name is provided in the constructor
+  public HDF5LongDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String datapath, String fieldName, int currentColumn) {
+    super(reader, columnWriter, datapath, fieldName, currentColumn);
+    // Get dimensions
+    long[] dimensions = reader.object().getDataSetInformation(datapath).getDimensions();
+    long[][] tempData;
+    if (dimensions.length == 2) {
+      tempData = transpose(reader.readLongMatrix(datapath));
+    } else {
+      tempData = transpose(reader.int64().readMDArray(datapath).toMatrix());
+    }
+    data = tempData[currentColumn];
+    rowWriter = makeWriter(columnWriter, fieldName, TypeProtos.MinorType.BIGINT, TypeProtos.DataMode.OPTIONAL);
+  }
+
+  public HDF5LongDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String fieldName, List tempListData) {
+    super(reader, columnWriter, null);
+    this.fieldName = fieldName;
+    data = new long[tempListData.size()];
+    for (int i = 0; i < tempListData.size(); i++) {
+      data[i] = (Long)tempListData.get(i);
+    }
+    rowWriter = makeWriter(columnWriter, fieldName, TypeProtos.MinorType.BIGINT, TypeProtos.DataMode.OPTIONAL);
+  }
+
+  public boolean write() {
+    if (counter > data.length) {
+      return false;
+    } else {
+      rowWriter.setLong(data[counter++]);
+      return true;
+    }
+  }
+
+  public boolean hasNext() {
+    return counter < data.length;
+  }
+
+  private long[][] transpose (long[][] array) {
```

Comment:
Looks like this transpose logic is the same except of types in arrays. Could be nice if we could find a way to extract logic into common method though.